### ISSUE TITLE
Flip/Trade entity

### ISSUE THREAD

**Issue by vikiival** [kodadot/loligo#29]
To make some trading views easier the optional path is to make a Trade entity like following

```graphql
type TradeEntity @entity {
  id: ID!
  nft: NFTEntity!
  createdAt: DateTime!
  closedAt: DateTime!
  initialPrice: BigInt!
  closingPrice: BigInt!
  realized: BigInt!
  percent: Int!
  profit: Boolean!
  issuer: String!
  buyer: String! 
}
```

May @daiagi can have more information since he was working on Flipper UI in Kodadot

**Comment by vikiival** [comment]
+ maybe add how much fee + royalty was paid?

**Comment by daiagi** [comment]
here is the data shape currently created on client side

getting all the flip events on a collection from squid is good enough, as the rest is simple summation
obv for a flip event we will need also the flipper id, which is the address that bought and then later sold (i.e the nft got bought from him/her/it) an nft
a person who got the nft for free (minter or a sendee) is not counted as flipper

```ts
export type FlipEvent = {
  nft: NFTExcludingEvents
  soldPrice: number
  soldTo: string
  sellTimeStamp: number
  boughtPrice: number
  profit: number
}

export type Flippers = {
  [identity: string]: {
    flips: FlipEvent[]
    owned: number
    totalBought: number
    totalsold: number
    bestFlip: number
    latestflipTimestamp: number
  }
}
```

**Comment by vikiival** [comment]
> getting all the flip events on a collection from squid is good enough

But not effective imho, you still need to iterate over buys or so

> flipper id, which is the address that bought and then later sold

Marked as `caller` in the schema 

> a person who got the nft for free (minter or a sendee) is not counted as flipper

Yup otherwise it would be always 100% profit

**Comment by daiagi** [comment]
>But not effective imho, you still need to iterate over buys or so

absolutely, getting it all nice and organized straight from Mr.squid would be ideal
